### ISSUE TITLE
Rstudio does not start

### ISSUE THREAD

**Issue by FredrikKarlssonSpeech** [humlab-speech/humlab-speech-deployment#38]

**Comment by johanvonboer** [comment]
Everything looks completely fine in the logs and you have an RStudio session running, could it be a client side issue? What browser were you using?

**Comment by FredrikKarlssonSpeech** [comment]
Maybe this is an issue of me just having the patience to wait. The session took forever to start. Related to the size of my database? I will close the issue for now and return to it if I see it again.

**Comment by johanvonboer** [comment]
Oh! Yes the size of the repo will definitely have an impact, since everything needs to be 'downloaded' from GitLab to the session container, and despite that it's being done on the internal docker network, this always takes some time.

This feels like a great example of one of those issues that pop up in real world testing that you just don't see in development... Not sure what to do about it right now, but it's good to keep this issue in mind when moving forward...

**Comment by FredrikKarlssonSpeech** [comment]
Ok, but once you have a jypyter notebook or Rstudio running, the container created will have everything there so that you can reconnect to it later (tomorrow), right? Without being forced to clone the repo again.

What possibilities are there for getting information that could feed a progress bar?

**Comment by johanvonboer** [comment]
Yes, if you don't shutdown the session/container you can just reconnect to it later, but of course there are problems with that as well, such as:

1) Unless you have clicked the save button in the portal site, or done a manual commit+push yourself from inside the session. Your changes only exists in the container, which will be lost during a reboot (which I just performed without realizing that you might be working on something - sorry!).

2) Its also more likely that you will end up with git conflicts if you are multiple people working on the same repo. And you might even shoot yourself in the foot a little bit by doing some edits in Jupyter, and some in RStudio and then you might have singlehandedly create git conflicts.

So there are issues here... I'm not saying this is a bad architecture, it has some upsides as well, but there certainly are problems with this approach that we need to consider and try to mitigate.

Yes, getting a progress bar would be great, might not be easy to do, but I'll have a look at it.

**Comment by johanvonboer** [comment]
I forgot to mention that leaving open containers of course also sucks up system resources on the server.

Maybe one way to mitigate some of these problems is to only allow 1 application at a time? So if you have launched RStudio, and you're back at the portal site, then you need to shutdown rstudio before you are allowed to launch jupyter or emuwebapp, worth considering perhaps?

**Comment by FredrikKarlssonSpeech** [comment]
Two observations / suggestions that together may make a solution?:

1) I think we could ask the user when the project is created whether to set up a "Streamlined (Transcribe->Valication->Analysis)" workflow, or a "Programmer friendly" workflow. We can then hide either Rstudio or jypyter from in the main interface and that way make sure that the two are not started at the same time

2) R will save the workspace and execute .Last  if it receives a SIGUSER1 it seems. https://stat.ethz.ch/R-manual/R-devel/library/base/html/Signals.html. There is an R package that makes you commit files to repositories (git2r) and if we defilne a .Last function in the .Rprofile we set up for a project, then the repo will be committed by R when it is quit by the user or when receiving the USER1 signal. 

I do see a potential for conflict, but we are then at the level of users who are brave enough to do large-scale alterations of transcriptions using R / and careless enough to have uncommitted transcription changes in the web app. Not sure how to fix such behavior.
